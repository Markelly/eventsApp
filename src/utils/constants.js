export const DISCOVER_URL = "https://app.ticketmaster.com/discovery/v2";
export const API_KEY = "4diZ7tJQArO9sxHiVWMASl5udVTAoI78";

export const EVENTS_PATH = "/events";
export const SUGGEST_PATH = "/suggest";

export const SORT_OPTIONS = [
  { text: "Upcoming events", value: "date,asc" },
  { text: "Name", value: "name,asc" },
  { text: "Relevance", value: "relevance,asc" },
  { text: "Distance", value: "distance,asc" }
];
