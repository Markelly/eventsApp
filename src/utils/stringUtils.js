import { EVENTS_PATH } from "./constants";

export const getEventsList = eventsList => {
  let event = {};
  let events = [];
  eventsList.forEach(item => {
    event.name = item.name;
    event.date = getDate(item);
    event.price = getPrice(item);
    events.push(event);
    event = {};
  });
  return events;
};

export const getPrice = item => {
  const min = item.priceRanges ? item.priceRanges[0].min : 0;
  const max = item.priceRanges ? item.priceRanges[0].max : 0;
  return item.priceRanges ? `€${min} - €${max}` : "Free";
};

export const getDate = item => {
  let formattedDate = "No date informed";
  const date = new Date(
    item.dates && item.dates.start && item.dates.start.localDate
  );
  if (date) {
    let weekday = getWeekDayName(date.getDay());
    let month = getMonthName(date.getMonth());
    let day = date.getDate();
    let year = date.getFullYear();
    formattedDate = `${weekday}, ${day} ${month} ${year}`;
  }
  return formattedDate;
};

export const getWeekDayName = day => {
  const days = [
    "Sunday",
    "Monday",
    "Tuesday",
    "Wednesday",
    "Thursday",
    "Friday",
    "Saturday"
  ];
  return days[day];
};

export const getMonthName = month => {
  const months = [
    "January",
    "February",
    "March",
    "April",
    "May",
    "June",
    "July",
    "August",
    "September",
    "October",
    "November",
    "December"
  ];
  return months[month];
};

export const isEventsPage = path => {
  return path === EVENTS_PATH;
};

export default {
  getEventsList,
  getDate,
  getPrice,
  getWeekDayName,
  getMonthName,
  isEventsPage
};
