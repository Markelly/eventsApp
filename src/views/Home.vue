<template>
  <div class="home">
    <div class="home__menu">
      <div
        v-if="isEventsRoute"
        class="home__menu__item"
        @click="goToSuggest"
        data-suggest-button
      >
        Browse suggestions
      </div>
      <div
        v-else
        class="home__menu__item"
        @click="goToEvents"
        data-events-button
      >
        See all events
      </div>
    </div>
  </div>
</template>

<script>
import { EVENTS_PATH, SUGGEST_PATH } from "../utils/constants";
import { isEventsPage } from "../utils/stringUtils";

export default {
  name: "Home",
  methods: {
    goToEvents() {
      this.$router.push(EVENTS_PATH);
    },
    goToSuggest() {
      this.$router.push(SUGGEST_PATH);
    }
  },
  computed: {
    isEventsRoute() {
      return isEventsPage(this.$route.path);
    }
  }
};
</script>

<style scoped lang="scss">
@import "../styles/lt-challenge";
</style>
