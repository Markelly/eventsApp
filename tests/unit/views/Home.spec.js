import Home from "../../../src/views/Home";
import { createLocalVue, shallowMount } from "@vue/test-utils";
import Router from "vue-router";
import { EVENTS_PATH, SUGGEST_PATH } from "../../../src/utils/constants";

describe("Home", () => {
  const localVue = createLocalVue();
  localVue.use(Router);
  let wrapper, mockRoute, spyRoutePush;

  beforeEach(() => {
    mockRoute = new Router({
      routes: [
        {
          path: EVENTS_PATH
        },
        {
          path: SUGGEST_PATH
        }
      ]
    });
    spyRoutePush = jest.spyOn(mockRoute, "push");
    mockRoute.push(EVENTS_PATH);
    wrapper = shallowMount(Home, {
      localVue,
      router: mockRoute,
      computed: {
        isEventRoute() {
          return true;
        }
      }
    });
  });

  test("should render the component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should render events page", () => {
    expect(wrapper.find("[data-suggest-button]")).toBeTruthy();
  });

  test("should render suggest page", () => {
    mockRoute.push(SUGGEST_PATH);
    expect(wrapper.find("[data-suggest-events]")).toBeTruthy();
  });

  test("should go to events page", () => {
    wrapper.vm.goToEvents();
    expect(spyRoutePush).toHaveBeenCalledWith(EVENTS_PATH);
  });

  test("should go to suggest page", () => {
    wrapper.vm.goToSuggest();
    expect(spyRoutePush).toHaveBeenCalledWith(SUGGEST_PATH);
  });
});
