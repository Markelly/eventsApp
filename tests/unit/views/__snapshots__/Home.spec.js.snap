// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render the component 1`] = `
<div class="home">
  <div class="home__menu">
    <div data-suggest-button="" class="home__menu__item">
      Browse suggestions
    </div>
  </div>
</div>
`;
